src/main/java/co/cask/common/AvroToStructuredTransformer.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.common;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.format.UnexpectedFormatException;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.hydrator.common.RecordConverter;
+import com.google.common.collect.Maps;
+import org.apache.avro.generic.GenericRecord;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Create StructuredRecords from GenericRecords
+ */
+public class AvroToStructuredTransformer extends RecordConverter<GenericRecord, StructuredRecord> {
+  private final Map<Integer, Schema> schemaCache = Maps.newHashMap();
+
+  public StructuredRecord transform(GenericRecord genericRecord) throws IOException {
```

Comment:
doesn't look like this is used

Reply:
Its an abstract method in `RecordConverter` https://github.com/caskdata/hydrator-plugins/blob/develop/hydrator-common/src/main/java/co/cask/hydrator/common/RecordConverter.java#L39